### PR TITLE
Improved SolarisbankAGPDFExtractor

### PR DESCRIPTION
**Pull request by kellerkind87**
- added SEPA-Überweisung to REMOVALS
- added Kartenvorgang to DEPOSITS

### CODE REVIEW

**Comment by buchen** [comment]
Hi @kellerkind87,

the change looks good (@Nirus2000 to comment on the PDF importer parts).
One word about the pull request: instead of *merging* the main branch into your feature branch, *rebase* the feature branch on the head of the main branch. That makes it much easier to review and then merge the changes.

**Comment by kellerkind87** [comment]
> Hi @kellerkind87,
> 
> the change looks good (@Nirus2000 to comment on the PDF importer parts). One word about the pull request: instead of _merging_ the main branch into your feature branch, _rebase_ the feature branch on the head of the main branch. That makes it much easier to review and then merge the changes.

yes indeed. i had some probs with git an eclipse.

**Comment by Nirus2000** [comment]
Hello @kellerkind87 
The test case does not contain all variables to be tested.

Since recently we have a very smart variant to write the test cases for Importer and to get you familiar with it, it would be great if you adapt your test case to the new variant.

Have a look at the [contributing rules](https://github.com/buchen/portfolio/blob/master/CONTRIBUTING.md#configure-eclipse) and complete your configuration of the favorites in Eclipse.

I have compiled an overview of the new test case for you here and the formatting should be adhered to in this way.
- [Buy & Sell](https://github.com/buchen/portfolio/blob/master/name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/baaderbank/BaaderBankPDFExtractorTest.java#L1031:L1059)
- [Dividende](https://github.com/buchen/portfolio/blob/777cd54cffb05dbd1493eb03b2d026c56098ffdd/name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/sbroker/SBrokerPDFExtractorTest.java#L1644:L1671)
- [Deposit, Removal, Interest, Fees for account transactions](https://github.com/buchen/portfolio/blob/777cd54cffb05dbd1493eb03b2d026c56098ffdd/name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/sbroker/SBrokerPDFExtractorTest.java#L2624:L2665)
- [NonImportable transactions](https://github.com/buchen/portfolio/blob/777cd54cffb05dbd1493eb03b2d026c56098ffdd/name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/fintechgroupbank/FinTechGroupBankPDFExtractorTest.java#L1344:L1375)

andthe others...

If you can, please adjust the [regular expressions](https://github.com/buchen/portfolio/blob/master/CONTRIBUTING.md#regular-expressions) to the contributing standards.

Have fun...

**Comment by kellerkind87** [comment]
hope this time i catche everything.

btw i like those new matchers!

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/solarisbank/SolarisbankAGPDFExtractorTest.java [hunk 1]
```diff
@@ -34,46 +34,19 @@ public void testGiroKontoauszug01()
...
-        assertThat(transaction.getDateTime(), is(LocalDateTime.parse("2022-10-26T00:00")));
-        assertThat(transaction.getAmount(), is(Values.Amount.factorize(100)));
-
-        item = iter.next();
-        transaction = (AccountTransaction) item.getSubject();
-        assertThat(transaction.getType(), is(AccountTransaction.Type.REMOVAL));
-        assertThat(transaction.getCurrencyCode(), is(CurrencyUnit.EUR));
-        assertThat(transaction.getDateTime(), is(LocalDateTime.parse("2022-10-28T00:00")));
-        assertThat(transaction.getAmount(), is(Values.Amount.factorize(16.10)));
+        assertThat(results.stream().filter(TransactionItem.class::isInstance).count(), is(4L));
```
Comment (Nirus2000) [inline name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/solarisbank/SolarisbankAGPDFExtractorTest.java:37]: remove this line

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/solarisbank/SolarisbankAGPDFExtractorTest.java [hunk 2]
```diff
@@ -34,46 +34,19 @@ public void testGiroKontoauszug01()
 
         assertThat(errors, empty());
```
Comment (Nirus2000) [inline name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/solarisbank/SolarisbankAGPDFExtractorTest.java:35]: Add this
`
        assertThat(errors, empty());
        assertThat(countSecurities(results), is(0L));
        assertThat(countBuySell(results), is(0L));
        assertThat(countAccountTransactions(results), is(9L));
        assertThat(results.size(), is(4));
        new AssertImportActions().check(results, CurrencyUnit.EUR);
`

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/solarisbank/SolarisbankAGPDFExtractorTest.java [hunk 3]
```diff
@@ -85,19 +58,10 @@ public void testGiroKontoauszug02()
...
 
-        assertThat(results.stream().filter(i -> i instanceof TransactionItem)
-                        .filter(i -> i.getSubject() instanceof AccountTransaction)
-                        .filter(i -> ((AccountTransaction) i.getSubject()).getType() == AccountTransaction.Type.DEPOSIT)
-                        .count(), is(1L));
+        assertThat(results.stream().filter(TransactionItem.class::isInstance).count(), is(1L));
```
Comment (Nirus2000) [inline name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/solarisbank/SolarisbankAGPDFExtractorTest.java:61]: remove this line and add this
`        assertThat(errors, empty());
        assertThat(countSecurities(results), is(0L));
        assertThat(countBuySell(results), is(0L));
        assertThat(countAccountTransactions(results), is(1L));
        assertThat(results.size(), is(1));
        new AssertImportActions().check(results, CurrencyUnit.EUR);`

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/solarisbank/SolarisbankAGPDFExtractorTest.java [hunk 4]
```diff
@@ -110,19 +74,10 @@ public void testGiroKontoauszug03()
...
         assertThat(errors, empty());
 
-        assertThat(results.stream().filter(i -> i instanceof TransactionItem)
-                        .filter(i -> i.getSubject() instanceof AccountTransaction)
-                        .filter(i -> ((AccountTransaction) i.getSubject()).getType() == AccountTransaction.Type.REMOVAL)
-                        .count(), is(1L));
-
-        Iterator<Extractor.Item> iter = results.stream().filter(i -> i instanceof TransactionItem).iterator();
-        Item item = iter.next();
+        assertThat(results.stream().filter(TransactionItem.class::isInstance).count(), is(1L));
```
Comment (Nirus2000) [inline name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/solarisbank/SolarisbankAGPDFExtractorTest.java:77]: remove and add this
```
        assertThat(errors, empty());
        assertThat(countSecurities(results), is(0L));
        assertThat(countBuySell(results), is(0L));
        assertThat(countAccountTransactions(results), is(1L));
        assertThat(results.size(), is(1));
        new AssertImportActions().check(results, CurrencyUnit.EUR);
```

File: name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/solarisbank/SolarisbankAGPDFExtractorTest.java [hunk 5]
```diff
@@ -132,37 +87,17 @@ public void testGiroKontoauszug04()
         List<Exception> errors = new ArrayList<>();
         List<Item> results = extractor.extract(loadFile("GiroKontoauszug04.txt"), errors);
         assertThat(errors, empty());
```
Comment (Nirus2000) [inline name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/solarisbank/SolarisbankAGPDFExtractorTest.java:89]: see at the others

File: name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/SolarisbankAGPDFExtractor.java [hunk 6]
```diff
@@ -81,7 +81,7 @@ private BiConsumer<AccountTransaction, ParsedData> assignmentsProvider()
             transaction.setDateTime(asDate(matcherMap.get(CONTEXT_KEY_DATE)));
             transaction.setAmount(asAmount(matcherMap.get(CONTEXT_KEY_AMOUNT)));
             transaction.setCurrencyCode(matcherMap.get(CONTEXT_KEY_CURRENCY));
-            transaction.setNote(matcherMap.get(CONTEXT_KEY_NOTE));
+            transaction.setNote(matcherMap.get(CONTEXT_KEY_NOTE).trim());
```
Comment (Nirus2000) [inline name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/SolarisbankAGPDFExtractor.java:84]: change to 
`transaction.setNote(trim(matcherMap.get(CONTEXT_KEY_NOTE)));`

look [here](https://github.com/buchen/portfolio/blob/777cd54cffb05dbd1493eb03b2d026c56098ffdd/name.abuchen.portfolio/src/name/abuchen/portfolio/util/TextUtil.java#L71:L90)

**Comment by buchen** [comment]
👍

**Pull request merged**